specifyweb/businessrules/migrations/0001_initial.py

```diff
@@ -0,0 +1,44 @@
+# Generated by Django 3.2.15 on 2023-09-30 17:35
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        ('specify', '__first__'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='UniquenessRule',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('isdatabaseconstraint', models.BooleanField(default=False)),
+                ('discipline', models.ForeignKey(db_column='DisciplineID', on_delete=django.db.models.deletion.PROTECT, to='specify.discipline')),
+                ('scope', models.ForeignKey(blank=True, db_column='splocalecontaineritemid', null=True, on_delete=django.db.models.deletion.PROTECT, to='specify.splocalecontaineritem')),
+            ],
+            options={
+                'db_table': 'uniquenessrule',
+            },
+        ),
+        migrations.CreateModel(
+            name='UniquenessRule_Splocalecontaineritem',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('splocalecontaineritem', models.ForeignKey(db_column='splocalecontaineritemid', on_delete=django.db.models.deletion.PROTECT, to='specify.splocalecontaineritem')),
+                ('uniquenessrule', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='businessrules.uniquenessrule')),
+            ],
+            options={
+                'db_table': 'uniquenessrule_splocalecontaineritem',
```

Comment:
There are two relationships to `splocalecontaineritem` from `uniquenessrule`: the many-to-many and the ForeignKey on uniquenessrule. 

The ForeignKey on uniquenessrule to splocalecontaineritem is to designate the field which acts as a "scope" for the rule. 
The many-to-many relationship is for all other fields of the uniquenessrule. Each additional field added to a uniqueness rule acts like an AND operation, where the field values are considered in conjuction when determining if they are unique or not. 

In production, an example of this rule is: Agent and Role in AccessionAgent must be unique to Accession.
This example would currently be represented in the database as new uniqueness rule which references the splocalecontaineritem (field) for the `AccessionAgent -> Accession` relationship as its scope and then reference the Agent and Role splocalecontaineritems on AccessionAgent in the `uniquenessrule_splocalecontaineritem` table. 

The Agent, Role, and Accesion fields can still be used in other uniqueness rules, hence the many-to-many relationship.  

Though, as stated in https://github.com/specify/specify7/pull/4215#issue-1996052310: 
> Note: Practically the distinction between a 'scope' and the 'fields' of a uniqueness rule is solely conceptual. They are the same as far as the operations are performed, and we could easily remove the scope field from uniquenessrule and instead have it as another to-many in uniquenessrule_splocalecontaineritem

Reply:
> The ForeignKey on uniquenessrule to splocalecontaineritem is to designate the field which acts as a "scope" for the rule.

This design has since been change to have the "scope" of the uniqueness rule be another entry within the many-to-many join table. It is differentiated from the other fields a rule contains by a boolean (bit value) field.